kafka_consumer/check.py

```diff
@@ -163,56 +299,76 @@ def _get_broker_offsets(self, instance):
-        except Exception:
-            self.log.exception('There was a problem collecting the high watermark offsets')
+        for topic, partitions in topics.iteritems():
+            # if no partitions are provided
+            # we're falling back to all available partitions (?)
+            if len(partitions) == 0:
+                partitions = cli.cluster.available_partitions_for_topic(topic)
+            topics_to_fetch[topic].update(partitions)
+
+
+        leader_tp = defaultdict(lambda: defaultdict(set))
+        for topic, partitions in topics_to_fetch.iteritems():
+            for partition in partitions:
+                partition_leader = cli.cluster.leader_for_partition(TopicPartition(topic, partition))
+                if partition_leader is not None and partition_leader > -1:
+                    leader_tp[partition_leader][topic].update([partition])
+
+        partitions_grouped_by_topic = defaultdict(list)
+
+        max_offsets = 1
+        for nodeId, tps in leader_tp.iteritems():
+            # Construct the OffsetRequest
+            request = OffsetRequest[0](
+                replica_id=-1,
+                topics=[
+                    (topic, [
+                        (partition, OffsetResetStrategy.LATEST, max_offsets) for partition in partitions])
+                    for topic, partitions in tps.iteritems()])
+
+            response = self._make_blocking_req(cli, request, nodeid=nodeId)
+            offsets, unled = self._process_highwater_offsets(request, instance, nodeId, response)
+            highwater_offsets.update(offsets)
+            topic_partitions_without_a_leader.extend(unled)
 
         return highwater_offsets, list(set(topic_partitions_without_a_leader))
 
+    def _report_consumer_metrics(self, highwater_offsets, consumer_offsets, unled_topic_partitions=[], tags=[]):
+        for (consumer_group, topic, partition), consumer_offset in consumer_offsets.iteritems():
+            # Report the consumer group offsets and consumer lag
+            if (topic, partition) not in highwater_offsets:
+                self.log.warn("[%s] topic: %s partition: %s was not available in the consumer "
+                              "- skipping consumer submission", consumer_group, topic, partition)
+                if (topic, partition) not in unled_topic_partitions:
+                    self.log.warn("Consumer group: %s has offsets for topic: %s "
+                        "partition: %s, but that topic partition doesn't actually "
+                        "exist in the cluster.", consumer_group, topic, partition)
+                continue
+
+            consumer_group_tags = ['topic:%s' % topic, 'partition:%s' % partition,
+                'consumer_group:%s' % consumer_group] + tags
+            self.gauge('kafka.consumer_offset', consumer_offset, tags=consumer_group_tags)
+
+            consumer_lag = highwater_offsets[(topic, partition)] - consumer_offset
+            if consumer_lag < 0:
```

Comment:
I was never a fan of this - I _always_ considered this to be the business of a monitor on the backend, it clutters the check with something we should probably be doing elsewhere. There was some push from the community to include this, but there was also some disagreement over its real value. If we all agree, and this hasn't been released in any form in any previous PR (can't remember off the top of my head), I'd favor removing it.